lib/board.ex

```diff
@@ -0,0 +1,38 @@
+defmodule Board do
+  @moduledoc """
+  The module that handles the Board logic for a Tic-Tac-Toe game
+  """
+
+  @spec setup_initial_board(non_neg_integer()) :: map()
+  def setup_initial_board(board_dimensions \\ 3) do
+    Map.new(1..board_dimensions * board_dimensions, fn position -> {position, "#{position}"} end)
+  end
+
+  @spec update_board(map(), String.t(), String.t()) :: map()
+  def update_board(unformatted_board, player_move, symbol) do
+    Map.replace(unformatted_board, String.to_integer(player_move), symbol)
+  end
+
+  @spec format_board_to_display(map()) :: String.t()
+  def format_board_to_display(unformatted_board) do
```

Comment:
This formatting stuff seems very command-line-centric, which makes me think it should belong to the command line module or a representation module.

Reply:
Hmmm, that is a good point and something I was debating before as well, I think Eli and I talked about this and I think we came to the conclusion that although the display (to the command line) requires formatting, the formatting itself was more general purpose to any visual medium (Browser, maybe a gui of some sort) where as the `CommandLine` `display` and `read_line` functions were strictly/directly related to the `CommandLine`. A representation module is not something we talked about though, which I would think could be a good idea, but I also feel like since the formatting of the board is so closely related to the board itself, it seems like a good place for it. This is a really helpful to think about regardless, so thank you for bringing this up 😄  I appreciate all the improvements/comments Steven! 🥳

Reply:
Sorry I probably should have elaborated on what I meant by "we can create different view...". pretty much what @elijahsullivan said here about the ability to support different UI(command line, web, etc) if needed without having to change `Board`.